Contribution Checklist (#273)
demisto/content-docs: docs/contributing/checklist.md

---
id: checklist
title: Contribution Checklist
---

This document includes a checklist that summarizes the list of files that you need in order to contribute to the Cortex XSOAR content repository. Please make sure you have everything before you open a Pull Request (unless stated otherwise).

Keep in mind that content packs can contain multiple types of entities, such as Integrations, Automations, Playbooks, Incident Types, Incident Fields, and so on.

> Depending on whether your content pack is aiming to be *certified* or not, there will be different requirements.

Reviewer: ```suggestion
Depending on whether your content pack is aiming to be [certified](https://xsoar.pan.dev/docs/partners/certification) or not, there will be different requirements.
```

Reviewer: @fvigo Why not referencing the certification document?

Author: sorry missed, this it's ok now

This article also includes a Pull Request checklist, that summarized everything you must do before and after opening a Pull Request on GitHub to contribute your pack.

Happy contributing!

# Content Pack checklist

> No matter you include in your content pack or whether it's going to be certified, the pack must include the following:

Reviewer: ```suggestion
No matter what you include in your content pack or whether it's going to be certified, the pack must include the following:
```

> - Pack Metadata file (i.e. `Packs/YourPackName/pack_metadata.json`) : the information about your content pack. It should be compiled with all the required information, as explained [here](packs-format).

Reviewer: ```suggestion
- Pack Metadata file (i.e. `Packs/YourPackName/pack_metadata.json`) : the information about your content pack. It should be compiled with all the required information, as explained [here](https://xsoar.pan.dev/docs/integrations/packs-formatt).
```

> - Pack Metadata file (i.e. `Packs/YourPackName/pack_metadata.json`) : the information about your content pack. It should be compiled with all the required information, as explained [here](packs-format).

Reviewer 2: fix all the relative links. 
`packs-format, pack-docs, release-notes, etc.`

- Pack README (i.e. `Packs/YourPackName/README.md`): the readme of the pack file. More information [here](pack-docs).
- Release Notes (i.e. `Packs/YourPackName/ReleaseNotes/1_0_1.md`): these are required only if you are updating an existing Content pack, not for the first. More information [here](release-notes).

> 

Reviewer: Are we planning to move all three of these docs under contributing section? Right now there are under `integrations/packs-format` path and under `Getting Started` in the sidebar.

Author: Let me fix the links first, we'll see the reorganization tomorrow.

- Command examples file (i.e. `Packs/YourPackName/Integrations/YourIntegrationName/command_examples`): needed to [autogenerate](integration-docs#command-examples) the documentation
- Unit tests file (i.e. `Packs/YourPackName/Integrations/YourIntegrationName/YourIntegrationName_test.py`): must be included to automatically test the code during the review phase. More details [here](unit-testing)
- Unit tests data (i.e. `Packs/YourPackName/Integrations/YourIntegrationName/test_data/*.json`): contain example responses from your product API and are used in [Unit tests](unit-testing). [Examples](https://github.com/demisto/content/tree/master/Packs/HelloWorld/Integrations/HelloWorld/test_data) from Hello World.

**Note**: if you use PowerShell and not Python, the extension of the code files will be `.ps1` instead of `.py`.

### Additional requirements Certified packs

If your integration is going to be *certified*, you also have the following requirements (optional for non-certified):

- Test Playbook (i.e. `Packs/YourPackName/TestPlaybooks/playbook-YourIntegrationName_Test.yml`): required for nightly builds. More information [here](test-playbooks)

> - Custom Incident Types, Fields, Classifiers, Mappers and Layouts: **if** your integration the ability to [*fetch incidents*](fetching-incidents), most likely you need to provide custom Incident Types and the related entities. This is usually covered during the Design phase: work with your Palo Alto Networks alliance contact if in doubt.

Reviewer 2: ```suggestion
- Custom Incident Types, Fields, Classifiers, Mappers and Layouts: **if** your integration has the ability to [*fetch incidents*](fetching-incidents), most likely you need to provide custom Incident Types and the related entities. This is usually covered during the Design phase: work with your Palo Alto Networks alliance contact if in doubt.
```

... and of course your adherence to our best practices and [code conventions](code-conventions) will be evaluated in a stricter way.

## Playbooks

If your pack contains at least a playbook, the playbook directory must contain the following files:

- Playbook file (i.e. `Packs/YourPackName/Playbooks/playbook-YourPlaybookName.yml`): if exported directly from the XSOAR UI, it must be formatted with `demisto-sdk format`.

> - README file (i.e. `Packs/YourPackName/Playbooks/playbook-YourPlaybookName.md`): documentation file for the Playbook, mostly [autogenerated](integration-docs#creating-documentation)

Reviewer 2: ```suggestion
- README file (i.e. `Packs/YourPackName/Playbooks/playbook-YourPlaybookName_README.md`): documentation file for the Playbook, mostly [autogenerated](integration-docs#creating-documentation)
```

**Note**: if your pack contains both integrations and scripts, you can use a single Test Playbook to test both.

## Widgets

If your pack contains at least a custom widget, you'll need:

- Widget JSON file (i.e. `Packs/YourPackName/Widgets/widget-YourWidgetName.json`): if exported directly from the XSOAR UI, it must be formatted with `demisto-sdk format`.

## Dashboards

If your pack contains at least a custom dashboard, you'll need:

- Dashboard JSON file (i.e. `Packs/YourPackName/Dashboard/dashboard-YourDashboardName.json`): if exported directly from the XSOAR UI, it must be formatted with `demisto-sdk format`.

# Checklist Table

The requirements above are also summarized in the following table:

|               Entity Type |                                                                          Non-certified and certified                                                                          | Certified only  |
|--------------------------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|---------------------------------|

> | Pack                      | <ul><li>Pack metadata</li><li>Pack readme</li><li>Release Notes</li></ul>                                                                                                                       | <ul><li>Test Playbook</li><li>Custom Incidents (if *fetch_incidents* is supported)</li></ul> |

Reviewer 2: Add here a row with `Design` and under `Certified only` mention that certified partner must implement most of the use cases from `Use Cases`
